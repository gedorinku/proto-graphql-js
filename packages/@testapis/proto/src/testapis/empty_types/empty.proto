syntax = "proto3";

package testapis.empty_types;

message EmptyMessage {
}
