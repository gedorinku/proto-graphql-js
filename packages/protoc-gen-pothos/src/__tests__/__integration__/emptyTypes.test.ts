import { testSchemaGeneration } from "../__helpers__/process.test.helper";

testSchemaGeneration("empty_types", "ts-proto", {
  schemaTests: [
    [
      "empty",
      {
        extraSchema: `
          import { builder } from "./builder";
          import { EmptyMessage$Ref } from "./testapis/empty_types/empty.pb.pothos";
          import { EmptyMessage } from "@testapis/ts-proto/lib/testapis/empty_types/empty";

          builder.queryField("test", (t) => t.field({
            type: EmptyMessage$Ref,
            resolve() {
              return EmptyMessage.fromPartial({});
            },
          }));
        `,
        testQuery: `
          query Test {
            test {
              _
            }
          }
        `,
      },
    ],
  ],
});
