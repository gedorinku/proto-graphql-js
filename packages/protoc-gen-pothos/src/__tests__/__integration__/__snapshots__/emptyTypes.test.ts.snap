// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema generation with ts-proto empty 1`] = `
Object {
  "data": Object {
    "test": Object {
      "_": true,
    },
  },
}
`;

exports[`schema generation with ts-proto empty: schema.graphql 1`] = `
"type EmptyMessage {
  \\"\\"\\"noop field\\"\\"\\"
  _: Boolean
}

input EmptyMessageInput {
  \\"\\"\\"noop field\\"\\"\\"
  _: Boolean
}

type Query {
  test: EmptyMessage!
}
"
`;
